packages/utils/src/internals/url.ts

```diff
@@ -0,0 +1,36 @@
+export type SearchParams = string | URLSearchParams | Record<string, string | number | boolean | null | undefined>;
+
+/**
+ * Appends search (query string) parameters to a URL, replacing the original value (if any).
+ *
+ * @param url The URL to append to.
+ * @param searchParams The search parameters to be appended.
+ */
+export function applySearchParams(url: URL, searchParams: SearchParams | undefined): void {
```

Comment:
is this intended to be used by downstream (in custom HTTP client implementations)? i see its used in `processHttpRequestOptions`, so if its only meant to be used from there, i'd mark it as `@internal`

Reply:
:shrug: sure, let's be conservative here